more_itertools/more.py

```diff
@@ -289,6 +319,32 @@ def __getitem__(self, index):
 
         return self._cache[index]
 
+    def index(self, value, start=0, stop=None):
```

Comment:
Did you consider delegating to `__getitem__` for the implementation? For example, call getitem with successively greater indices until we find a matching item. It would get us down to a single `for` loop and relieve us of cache maintenance duties here.

Reply:
Oh, good call - I had it in my head that this wouldn't work, but it does!